controllers/managedocs_controller_test.go

```diff
@@ -973,6 +964,73 @@ var _ = Describe("ManagedOCS controller", func() {
 					utils.WaitForResource(k8sClient, ctx, alertRelabelConfigSecretTemplate.DeepCopy(), timeout, interval)
 				})
 			})
+			When("the kube-rbac-proxy ConfigMap is modified", func() {
+				It("should revert the changes and bring the resource back to its managed state", func() {
+					// Get an updated configMap
+					kubeRBACConfigMap := prometheusKubeRBACPoxyConfigMapTemplate.DeepCopy()
+					kubeRBACConfigMapKey := utils.GetResourceKey(kubeRBACConfigMap)
+					Expect(k8sClient.Get(ctx, kubeRBACConfigMapKey, kubeRBACConfigMap)).Should(Succeed())
+
+					// Update to the configMap with dummy values
+					data := kubeRBACConfigMap.Data
+					kubeRBACConfigMap.Data = map[string]string{
+						"test-key": "test-value",
+					}
+					Expect(k8sClient.Update(ctx, kubeRBACConfigMap)).Should(Succeed())
+
+					// Wait for the configMap changes to be reverted
+					Eventually(func() map[string]string {
+						kubeRBACConfigMap := prometheusKubeRBACPoxyConfigMapTemplate.DeepCopy()
+						Expect(k8sClient.Get(ctx, kubeRBACConfigMapKey, kubeRBACConfigMap)).Should(Succeed())
+						return kubeRBACConfigMap.Data
+					}, timeout, interval).Should(Equal(data))
+				})
+			})
+			When("the kube-rbac-proxy ConfigMap resource is deleted", func() {
+				It("should create a new kube-rbac-proxy ConfigMap in the namespace", func() {
+					// Delete the NetworkPolicy resource
+					Expect(k8sClient.Delete(ctx, prometheusKubeRBACPoxyConfigMapTemplate.DeepCopy())).Should(Succeed())
+
+					// Wait for the NetworkPolicy to be recreated
+					utils.WaitForResource(k8sClient, ctx, prometheusKubeRBACPoxyConfigMapTemplate.DeepCopy(), timeout, interval)
+				})
+			})
+			When("the prometheus service is modified", func() {
+				It("should revert the changes and bring the resource back to its managed state", func() {
+					// Get an updated prometheus service
+					prometheusService := prometheusServiceTemplate.DeepCopy()
+					prometheusServiceKey := utils.GetResourceKey(prometheusService)
+					Expect(k8sClient.Get(ctx, prometheusServiceKey, prometheusService)).Should(Succeed())
+
+					// Update to empty spec
+					spec := prometheusService.Spec
+					prometheusService.Spec = corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{
+							{
+								Port: 1234,
+							},
+						},
+						ClusterIP: spec.ClusterIP,
+					}
```

Comment:
No need to add mock data, we just want to verify that it is reconciling back. An empty object will work just fine
```suggestion
					prometheusService.Spec = corev1.ServiceSpec{}
```

Reply:
It can't be done for service because ports is a required field and clusterIP is an immutable field. So we need to add a dummy port and keep the clusterID as whatever it was before.

Reply:
@bindrad Ok, please add a comment explaining exactly that